Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/serverexplore/hdinsightnode/StorageAccountNode.java

```diff
@@ -23,109 +23,69 @@
-            }
-            if (permissions != null) {
-                access = permissions.getPublicAccess().toString();
-            }
-            String name = container.getName();
-            String eTag = null;
-
-            String uri = container.getUri().toString();
-            Calendar lastModified = new GregorianCalendar();
-            BlobContainerProperties properties = container.getProperties();
-
-            if (properties != null) {
-                eTag = properties.getEtag();
-                lastModified.setTime(properties.getLastModified());
-            }
-            return new BlobContainer(name, uri, eTag, lastModified, access);
-        });
-    }
-
-    @Override
-    protected void refreshItems() {
-        try {
-            if (storageAccount.getAccountType() == StorageAccountType.BLOB) {
-                HDStorageAccount blobStorageAccount = (HDStorageAccount) storageAccount;
-                String defaultContainer = blobStorageAccount.getDefaultContainer();
-                final String connectionString = ((HDStorageAccount) storageAccount).getConnectionString();
-                getBlobContainers(connectionString).forEach(blobContainer -> {
-                    addChildNode(new BlobContainerNode(this, blobStorageAccount, blobContainer, !StringHelper.isNullOrWhiteSpace(defaultContainer) && defaultContainer.equals(blobContainer.getName())));
-                });
-            } else {
-                StringBuilder sb = new StringBuilder();
-                sb.append("Can't refresh the storage account since unsupported storage account type: " + storageAccount.getAccountType() + "\n");
-                sb.append("Account name: " + storageAccount.getName() + "\n");
-                sb.append("Subscription ID: " + storageAccount.getSubscriptionId() + "\n");
-                sb.append("Default storage schema: " + storageAccount.getDefaultStorageSchema() + "\n");
-                sb.append("Default container or root path: " + storageAccount.getDefaultContainerOrRootPath() + "\n");
-                log().warn(sb.toString());
-            }
-        } catch (Exception ex) {
-            String exceptionMsg = ex.getCause() == null ? "" : ex.getCause().getMessage();
-            String errorHint =  String.format("Failed to load storage account %s. ", storageAccount.getName());
-            log().warn(errorHint + ExceptionUtils.getStackTrace(ex));
-
-            DefaultLoader.getUIHelper().showError(errorHint + exceptionMsg, "HDInsight Explorer");
+    protected void loadAdditionalActions() {
+        if (clusterDetail instanceof ClusterDetail) {
+            addAction("Open Storage in Azure Management Portal", new NodeActionListener() {
+                @Override
+                protected void actionPerformed(NodeActionEvent e) throws AzureCmdException {
+                    try {
+                        final AzureManager azureManager = AuthMethodManager.getInstance().getAzureManager();
```

Comment:
Shall we use openResourcesInPortal(HasId , String)` in `Node` here? So that node could share codes to open resources in portal

Reply:
Resolved in the new iteration.